relay-event-normalization/src/event.rs

```diff
@@ -29,288 +19,133 @@ use relay_event_schema::protocol::{
-                Ok(())
-            } else {
-                meta.add_error(ErrorKind::InvalidData);
-                Err(ProcessingAction::DeleteValueSoft)
-            }
-        })?;
-        processor::apply(&mut event.environment, |environment, meta| {
-            if crate::validate_environment(environment).is_ok() {
-                Ok(())
-            } else {
-                meta.add_error(ErrorKind::InvalidData);
-                Err(ProcessingAction::DeleteValueSoft)
-            }
-        })?;
-
-        // Default required attributes, even if they have errors
-        normalize_logentry(&mut event.logentry, meta)?;
-        normalize_release_dist(event)?; // dist is a tag extracted along with other metrics from transactions
-        normalize_timestamps(
-            event,
-            meta,
-            self.config.received_at,
-            self.config.max_secs_in_past,
-            self.config.max_secs_in_future,
-        )?; // Timestamps are core in the metrics extraction
-        normalize_event_tags(event)?; // Tags are added to every metric
-
-        // TODO: Consider moving to store normalization
-        if self.config.device_class_synthesis_config {
-            normalize_device_class(event);
-        }
-        light_normalize_stacktraces(event)?;
-        normalize_exceptions(event)?; // Browser extension filters look at the stacktrace
-        normalize_user_agent(event, self.config.normalize_user_agent); // Legacy browsers filter
-        normalize_measurements(
+    // Default required attributes, even if they have errors
+    normalize_logentry(&mut event.logentry, meta)?;
+    normalize_release_dist(event)?; // dist is a tag extracted along with other metrics from transactions
+    normalize_timestamps(
+        event,
+        meta,
+        config.received_at,
+        config.max_secs_in_past,
+        config.max_secs_in_future,
+    )?; // Timestamps are core in the metrics extraction
+    normalize_event_tags(event)?; // Tags are added to every metric
+
+    // TODO: Consider moving to store normalization
+    if config.device_class_synthesis_config {
+        normalize_device_class(event);
+    }
+    light_normalize_stacktraces(event)?;
+    normalize_exceptions(event)?; // Browser extension filters look at the stacktrace
+    normalize_user_agent(event, config.normalize_user_agent); // Legacy browsers filter
+    normalize_measurements(
+        event,
+        config.measurements.clone(),
+        config.max_name_and_unit_len,
+    ); // Measurements are part of the metric extraction
+    normalize_performance_score(event, config.performance_score);
+    normalize_breakdowns(event, config.breakdowns_config); // Breakdowns are part of the metric extraction too
+
+    // Some contexts need to be normalized before metrics extraction takes place.
+    processor::apply(&mut event.contexts, normalize_contexts)?;
+
+    if config.light_normalize_spans && event.ty.value() == Some(&EventType::Transaction) {
```

Comment:
Same here, does it make sense to rename this config option and remove `light_normalize_...` from it?